src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Recipes/ElasticIndexRebuildStep.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Threading.Tasks;
+using OrchardCore.BackgroundJobs;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Search.Elasticsearch.Core.Services;
+
+namespace OrchardCore.Search.Elasticsearch.Core.Recipes
+{
+    /// <summary>
+    /// This recipe step rebuilds a lucene index.
+    /// </summary>
+    public class ElasticIndexRebuildStep : IRecipeStepHandler
+    {
+        private readonly ElasticIndexSettingsService _elasticIndexSettingsService;
+        private readonly ElasticIndexingService _elasticIndexingService;
+
+        public ElasticIndexRebuildStep(
+            ElasticIndexSettingsService elasticIndexSettingsService,
+            ElasticIndexingService elasticIndexingService
+            )
+        {
+            _elasticIndexSettingsService = elasticIndexSettingsService;
+            _elasticIndexingService = elasticIndexingService;
+        }
+
+        private async Task RebuildIndexAsync(string indexName)
+        {
+            var elasticIndexSettings = await _elasticIndexSettingsService.GetSettingsAsync(indexName);
+
+            if (elasticIndexSettings != null)
+            {
+                await _elasticIndexingService.RebuildIndexAsync(elasticIndexSettings);
+                await _elasticIndexingService.ProcessContentItemsAsync(indexName);
+            }
+        }
+
+        public async Task ExecuteAsync(RecipeExecutionContext context)
+        {
+            if (!String.Equals(context.Name, "elastic-index-rebuild", StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            var indices = context.Step["Indices"];
+
+            if (indices != null)
+            {
+                await HttpBackgroundJob.ExecuteAfterEndOfRequestAsync("elastic-index-rebuild", async (shellScope) => {
```

Comment:
The only issue I have so far is that this part noop.
@jtkech

Reply:
It works when it is executed as part of a Deployment plan. But it is not when executed from the Recipe list in the admin UI.

Reply:
Yes, you can't do that as you will be in a different scope, you have to use the passed scope to the delegate, this to resolve service(s) `scope.Services.GetRequiredService<>()` and then use this service(s) to do the job.

Hmm here looks like you would have to resolve ElasticIndexSettingsService and ElasticIndexingService and then do what RebuildIndexAsync does here. Or maybe you already have a service that injects these 2 above services and that have an interface method that does exactly the same.